File: sdks/javascript/test_sdk.js
/**
 * Test JavaScript/TypeScript SDK
 */

// Since we're testing the raw TS file, we'll use a simple Node.js test
// In production, this would be compiled first

const fs = require('fs');
const path = require('path');

console.log('==================================================');
console.log('  JAVASCRIPT SDK TEST');
console.log('==================================================');

let passed = 0;
let failed = 0;

function test(name, fn) {
    try {
        fn();
        console.log(`  [PASS] ${name}`);
        passed++;
    } catch (e) {
        console.log(`  [FAIL] ${name}: ${e.message}`);
        failed++;
    }
}

// Read and parse the TypeScript file to verify structure
const sdkPath = path.join(__dirname, 'src', 'index.ts');
const sdkContent = fs.readFileSync(sdkPath, 'utf-8');

console.log('\nTesting TypeScript SDK structure...\n');

// Test exports exist
test('RiskLevel type exported', () => {
    if (!sdkContent.includes("export type RiskLevel")) throw new Error('Missing');
});

test('Chain type exported', () => {
    if (!sdkContent.includes("export type Chain")) throw new Error('Missing');
});

test('AlertType type exported', () => {
    if (!sdkContent.includes("export type AlertType")) throw new Error('Missing');
});

test('RiskAssessment interface exported', () => {
    if (!sdkContent.includes("export interface RiskAssessment")) throw new Error('Missing');
});

test('ChainShield class exported', () => {
    if (!sdkContent.includes("export class ChainShield")) throw new Error('Missing');
});

test('ChainShieldError class exported', () => {
    if (!sdkContent.includes("export class ChainShieldError")) throw new Error('Missing');
});

test('AuthenticationError class exported', () => {
    if (!sdkContent.includes("export class AuthenticationError")) throw new Error('Missing');
});

test('RateLimitError class exported', () => {
    if (!sdkContent.includes("export class RateLimitError")) throw new Error('Missing');
});

// Test methods exist
console.log('\nTesting SDK methods...\n');

test('analyze method exists', () => {
    if (!sdkContent.includes("async analyze(")) throw new Error('Missing');
});

test('analyzeBatch method exists', () => {
    if (!sdkContent.includes("async analyzeBatch(")) throw new Error('Missing');
});

test('isSanctioned method exists', () => {
    if (!sdkContent.includes("async isSanctioned(")) throw new Error('Missing');
});

test('isHighRisk method exists', () => {
    if (!sdkContent.includes("async isHighRisk(")) throw new Error('Missing');
});

test('registerWebhook method exists', () => {
    if (!sdkContent.includes("async registerWebhook(")) throw new Error('Missing');
});

test('health method exists', () => {
    if (!sdkContent.includes("async health(")) throw new Error('Missing');
});

// Test chains
console.log('\nTesting chain support...\n');

const chains = ['ethereum', 'polygon', 'arbitrum', 'bsc', 'optimism', 'base', 'avalanche', 'fantom', 'zksync', 'bitcoin', 'solana'];
chains.forEach(chain => {
    test(`Chain ${chain} supported`, () => {
        if (!sdkContent.includes(`'${chain}'`)) throw new Error('Missing');
    });
});

// Summary
console.log('\n==================================================');
console.log(`  RESULT: ${passed}/${passed + failed} tests passed`);
console.log('==================================================');

process.exit(failed > 0 ? 1 : 0);

File: sdks/python/test_sdk.py
"""
Test Python SDK - Verify all components work
"""

import sys
sys.path.insert(0, 'd:/project/chainshield/sdks/python')

def test_imports():
    """Test all imports work."""
    print("Testing Python SDK imports...")
    
    try:
        from chainshield import (
            ChainShield,
            RiskAssessment,
            RiskLevel,
            Chain,
            ChainShieldError,
            AuthenticationError,
            RateLimitError,
        )
        print("  [PASS] All imports successful")
        return True
    except ImportError as e:
        print(f"  [FAIL] Import error: {e}")
        return False


def test_models():
    """Test model classes."""
    print("\nTesting models...")
    
    try:
        from chainshield.models import RiskAssessment, RiskLevel, Chain
        
        # Test RiskLevel enum
        assert RiskLevel.LOW.value == "LOW"
        assert RiskLevel.CRITICAL.value == "CRITICAL"
        print("  [PASS] RiskLevel enum works")
        
        # Test Chain enum
        assert Chain.ETHEREUM.value == "ethereum"
        assert Chain.BITCOIN.value == "bitcoin"
        print("  [PASS] Chain enum works")
        
        # Test RiskAssessment from_dict
        data = {
            "address": "0x123",
            "chain": "ethereum",
            "risk_score": 45.5,
            "risk_level": "MEDIUM",
            "blocked": False,
            "factors": ["test factor"]
        }
        assessment = RiskAssessment.from_dict(data)
        
        assert assessment.address == "0x123"
        assert assessment.risk_score == 45.5
        assert not assessment.is_high_risk
        assert not assessment.is_sanctioned
        print("  [PASS] RiskAssessment dataclass works")
        
        # Test high risk detection
        data["risk_level"] = "HIGH"
        data["risk_score"] = 85.0
        high_risk = RiskAssessment.from_dict(data)
        assert high_risk.is_high_risk
        print("  [PASS] is_high_risk property works")
        
        return True
    except Exception as e:
        print(f"  [FAIL] Model error: {e}")
        return False


def test_exceptions():
    """Test exception classes."""
    print("\nTesting exceptions...")
    
    try:
        from chainshield.exceptions import (
            ChainShieldError,
            AuthenticationError,
            RateLimitError,
            ValidationError,
        )
        
        # Test base error
        err = ChainShieldError("Test error", status_code=500)
        assert err.message == "Test error"
        assert err.status_code == 500
        print("  [PASS] ChainShieldError works")
        
        # Test auth error
        auth_err = AuthenticationError("Invalid key")
        assert isinstance(auth_err, ChainShieldError)
        print("  [PASS] AuthenticationError inheritance works")
        
        # Test rate limit error
        rate_err = RateLimitError("Too many requests", retry_after=60)
        assert rate_err.retry_after == 60
        print("  [PASS] RateLimitError works")
        
        return True
    except Exception as e:
        print(f"  [FAIL] Exception error: {e}")
        return False


def test_client_creation():
    """Test client can be created."""
    print("\nTesting client creation...")
    
    try:
        from chainshield.client import ChainShield
        from chainshield.exceptions import AuthenticationError
        
        # Test valid API key
        client = ChainShield(api_key="cs_test_key_123")
        assert client.api_key == "cs_test_key_123"
        print("  [PASS] Client created with valid key")
        
        # Test invalid API key format
        try:
            ChainShield(api_key="invalid_key")
            print("  [FAIL] Should have raised AuthenticationError")
            return False
        except AuthenticationError:
            print("  [PASS] Invalid key rejected correctly")
        
        # Test empty API key
        try:
            ChainShield(api_key="")
            print("  [FAIL] Should have raised AuthenticationError")
            return False
        except AuthenticationError:
            print("  [PASS] Empty key rejected correctly")
        
        return True
    except Exception as e:
        print(f"  [FAIL] Client error: {e}")
        return False


def main():
    print("=" * 50)
    print("  PYTHON SDK TEST")
    print("=" * 50)
    
    results = [
        test_imports(),
        test_models(),
        test_exceptions(),
        test_client_creation(),
    ]
    
    passed = sum(results)
    total = len(results)
    
    print("\n" + "=" * 50)
    print(f"  RESULT: {passed}/{total} tests passed")
    print("=" * 50)
    
    return 0 if all(results) else 1


if __name__ == "__main__":
    sys.exit(main())
